changefeedccl: add version gate for new changefeed options

sherman-grewal opened the pull request:
changefeedccl: add version gates for new changefeed options

Add a version gate for new changefeed options such as
end_time, initial_scan_only, and adding a string value
to initial_scan

Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/79500)
<!-- Reviewable:end -->

miretskiy commented on pkg/ccl/changefeedccl/alter_changefeed_stmt.go:
```diff
@@ -45,6 +46,10 @@ func alterChangefeedPlanHook(
 	}
 
+	if !p.ExecCfg().Settings.Version.IsActive(ctx, clusterversion.EnableNewChangefeedOptions) {
+		return nil, nil, nil, false, errors.Newf("alteration of changefeeds are not supported until upgrade to version %s is finalized", clusterversion.EnableNewChangefeedOptions.String())
```
how about `ALTER CHANGEFEED statement is not supported....`

miretskiy commented on pkg/ccl/changefeedccl/changefeed_stmt.go:
```diff
@@ -80,6 +81,10 @@ func changefeedPlanHook(
 		return nil, nil, nil, false, nil
 	}
 
+	if !p.ExecCfg().Settings.Version.IsActive(ctx, clusterversion.EnableNewChangefeedOptions) {
```
This is too wide.  We should only return this error if you use new option(s).

HonoreDB reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/79500)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @gh-casper, @miretskiy, and @sherman-grewal)

___
*[pkg/ccl/changefeedccl/alter_changefeed_stmt.go, line 50 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/79500#-MzzCJpKI6kODqt3uzcq-r1-50:-MzzFPVJAenZllP9vZ_J:b-f57c2q) ([raw file](https://github.com/cockroachdb/cockroach/blob/f45b8852285b17371d56fb5f3f47bd7fe7a89058/pkg/ccl/changefeedccl/alter_changefeed_stmt.go#L50)):*
<details><summary><i>Previously, miretskiy (Yevgeniy Miretskiy) wrote…</i></summary><blockquote>

how about `ALTER CHANGEFEED statement is not supported....`
</blockquote></details>

Also, "version %s or higher"


<!-- Sent from Reviewable.io -->

Commit:
changefeedccl: add version gates for new changefeed options
Add a version gate for new changefeed options such as
end_time, initial_scan_only, and adding a string value
to initial_scan

Release note: none
